source/posix/system_info.c

```diff
@@ -191,7 +191,15 @@ int s_parse_symbol(const char *symbol, void *addr, struct aws_stack_frame_info *
 
     long function_len = (open_paren && close_paren) ? close_paren - open_paren - 1 : 0;
     if (function_len > 0) { /* dynamic symbol was found */
-        strncpy(frame->function, open_paren + 1, function_len);
+        /* there might be (<function>+<addr>) or just (<function>) */
+        const char *function_start = open_paren + 1;
+        const char *plus = strstr(function_start, "+");
+        const char *function_end = (plus) ? plus : close_paren;
+        if (function_end > function_start) {
+            strncpy(frame->function, function_start, function_end - function_start);
+        } else if (plus) {
+            strncpy(frame->addr, plus + 1, close_paren - plus - 1);
```

Comment:
Doesn't this need a closing 0 byte?  You've already unconditionally copied the [] address into a buffer on line 190 and this is likely shorter

Reply:
Good catch.